lxljs/vocab.js

```diff
@@ -453,22 +450,16 @@ export function getProperties(classId, vocabClasses, vocabProperties, context) {
   if (termObj.allowedProperties) {
     return termObj.allowedProperties;
   }
-  vocabProperties.forEach((prop) => {
-    const domainList = getDomainList(prop, vocabProperties, context);
-    let domainListWithSubClasses = [];
-    for (let i = 0; i < domainList.length; i++) {
-      domainListWithSubClasses = domainListWithSubClasses.concat(
-        getSubClassChain(domainList[i], vocabClasses, context),
-      );
-    }
-    for (const domain of domainListWithSubClasses) {
-      if (domain === classId) {
-        props.push(prop);
+
+  const allowedProperties = Array.from(vocabProperties.values())
+    .filter((prop) => {
+      if (getPropertiesList(prop, vocabProperties, context).includes(classId)) {
+        return true;
       }
-    }
-  });
-  termObj.allowedProperties = props;
-  return props;
+      return false;
+    });
+  termObj.allowedProperties = allowedProperties;
```

Comment:
Should `termObj` be mutated here? There seems to be several cases of mutations which makes it difficult to trace the data flow.

Reply:
Agreed; let's discuss improvements here!

Reply:
I think that the main gripe I have (coming mainly from a functional programming background) is when a function used for getting data (as in `getProperties`) is also setting data elsewhere. In my world that should be separate concerns.

I'm currently in favor of manually appending the data for each class inside https://github.com/libris/lxlviewer/blob/b144ccf2f55df8e8ed8cb118ba76337f2baa8392/nuxt-app/src/store/index.js#L188-L210 if there are any related properties using `domain`/`domainIncludes` or `range`/`rangeIncludes`... and not relying on `getProperties` to populate the vocabClasses store.